processor/k8sattributesprocessor/processor.go

```diff
@@ -33,6 +33,9 @@ type kubernetesprocessor struct {
 	telemetrySettings      component.TelemetrySettings
 	logger                 *zap.Logger
 	apiConfig              k8sconfig.APIConfig
+	k8sClientProvider      kube.APIClientsetProvider
+	kcProvider             kube.ClientProvider
+	informerProviders      *kube.InformerProviders
```

Comment:
If the `informerProviders` is a nested field of `kubernetesprocessor`, how could we ensure the local caches of informer can be shared among all the kubernetes processor object?

Reply:
The actual providers are just functions, and their implementations can pull from a shared pool of informers. The `informerProviders` struct just contains all the providers potentially necessary for the processor to function. It only exists for convenience, so we don't have to pass around four provider functions everywhere.